### PR TITLE
Add "shortcodes" and "embeds" endpoints

### PR DESCRIPTION
**Pull request by TooTallNate**
These are publicly documented now, so we can include them here.

/cc @retrofox

### CODE REVIEW

File: lib/site.js [hunk 1]
```diff
@@ -239,6 +239,68 @@ Site.prototype.tag = function (slug) {
...
 
 /**
+ * Get a list of shortcodes available on a site.
+ *
+ * Note: The current user must have publishing access.
+ *
+ * @api public
+ */
+
+Site.prototype.shortcodes = function (query, fn) {
```
Comment (retrofox) [inline lib/site.js]: To be consistent we should rename this one to `shortcodesList` such as `postsList`, `usersList`, etc.

Comment (TooTallNate) [inline lib/site.js]: :+1: I'll update.

Comment (retrofox) [inline lib/site.js]: I think we can use `resource` array to auto generar that method

**Comment by retrofox** [comment]
@TooTallNate we can use `resources` array to autogenerate shortcodesList method https://github.com/Automattic/wpcom.js/blob/master/lib/site.js#L20

**Pull request merged**